engine/src/flutter/lib/web_ui/dev/generate_scene_test.dart

```diff
@@ -0,0 +1,257 @@
+    case 'colorFilter':
+      emitColorFilterOperation(operation, indent);
+    case 'imageFilter':
+      emitImageFilterOperation(operation, indent);
+    case 'offset':
+      emitOffsetOperation(operation, indent);
+    case 'opacity':
+      emitOpacityOperation(operation, indent);
+    case 'transform':
+      emitTransformOperation(operation, indent);
+    case 'shaderMask':
+      emitShaderMaskOperation(operation, indent);
+    default:
+      throw ArgumentError('invalid operation type: ${operation['type']}');
+  }
+}
+
+void emitLayer(Map<String, Object?> command, String indent) {
+  final layer = command['layer']! as Map<String, Object?>;
+  print('$indent{');
+  final String innerIndent = '  $indent';
+  emitOperation(layer['operation'], innerIndent);
+  emitCommands(layer['commands'], innerIndent);
+  print('${innerIndent}builder.pop();');
+  print('$indent}');
+}
+
+void emitPlatformView(Map<String, Object?> command, String indent) {
+  final localBounds = command['localBounds']! as Map<String, Object?>;
+  final left = localBounds['left']! as double;
+  final top = localBounds['top']! as double;
+  final right = localBounds['right']! as double;
+  final bottom = localBounds['bottom']! as double;
+  print(
+    '${indent}builder.addPlatformView(1, offset: const ui.Offset($left, $top), width: ${right - left}, height: ${bottom - top});',
+  );
+}
+
+void emitPicture(Map<String, Object?> command, String indent) {
+  print(
+    '${indent}builder.addPicture(${offsetAsString(command['offset'])}, drawPicture(${rectAsString(command['localBounds'])}, ${getColorAsString()}));',
+  );
+}
+
+void emitCommands(Object? commands, String indent) {
+  commands as List<Object?>?;
+  commands!;
+  for (final Object? command in commands) {
+    command as Map<String, Object?>?;
+    command!;
+    switch (command['type']) {
+      case 'picture':
+        emitPicture(command, indent);
+      case 'platformView':
+        emitPlatformView(command, indent);
+      case 'layer':
+        emitLayer(command, indent);
+    }
+  }
+}
+
+int main(List<String> args) {
+  if (args.length != 1) {
+    stderr.write('Usage: dart generate_scene_test.dart <path_to_json>.\n');
+    return 1;
+  }
+
+  final jsonFile = File(args[0]);
+  if (!jsonFile.existsSync()) {
+    stderr.write('Json file at path ${jsonFile.path} not found.\n');
+  }
+
+  final fileString = jsonFile.readAsStringSync();
+  final sceneJson = jsonDecode(fileString) as Map<String, Object?>;
+  final rootLayer = sceneJson['rootLayer']! as Map<String, Object?>;
+  print('''
+import 'dart:typed_data';
+import 'package:ui/ui.dart' as ui;
+
+ui.Picture drawPicture(ui.Rect bounds, ui.Color color) {
+  final recorder = ui.PictureRecorder();
+  final canvas = ui.Canvas(recorder);
+  canvas.drawRect(bounds, ui.Paint()..color = color);
+  return recorder.endRecording();
+}
+
+ui.Scene buildScene() {
+  final ui.SceneBuilder builder = ui.SceneBuilder();''');
+  emitCommands(rootLayer['commands'], '  ');
```

Comment:
Should we add a comment here saying these commands were generated from dumped info, etc?